src/Sarif/ConsoleLogger.cs

```diff
@@ -10,17 +10,46 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public class ConsoleLogger : IAnalysisLogger
     {
-        public ConsoleLogger(bool verbose)
+        public ConsoleLogger(bool verbose, string toolName)
         {
             Verbose = verbose;
+            _toolName = toolName.ToUpperInvariant();
         }
 
+        private StringBuilder _capturedOutput;
+
+        public bool CaptureOutput { get; set; }
+
+        public string CapturedOutput
+        {
+            get
+            {
+                if (_capturedOutput != null)
+                {
+                    return _capturedOutput.ToString();
+                }
+                return null;
```

Comment:
Or just `return _capturedOutput?.ToString();`

Reply:
Or write the whole thing on one line (saves 10 lines!):
```
    public string CapturedOutput => _capturedOutput?.ToString();
```

---
In reply to: [303172007](https://github.com/microsoft/sarif-sdk/pull/1574#discussion_r303172007) [](ancestors = 303172007)